dispatcher.c

```diff
@@ -933,7 +933,7 @@ dispatch_connection(connection *conn, dispatcher *self, struct timeval start)
 						router_route(self->rtr,
 							conn->dests, &conn->destlen, CONN_DESTS_SIZE,
 							conn->srcaddr,
-							conn->metric, firstspace, self->id - 1));
+							conn->metric, strchr(firstspace, ' '), self->id - 1));
```

Comment:
firstspace is a space, so why would you search for a space here?

Reply:
That's no so in case of tagged metric. Lets take `cpu_load;host=ex1;cpu=2 4 1579606658` as example. In cycle at this line https://github.com/grobian/carbon-c-relay/blob/4ecb19330f063ca8d851066834a82f9a5caf0787/dispatcher.c#L973 we found first semicolon in metric. What happends next...
```
} else if (search_tags && *p == ';') {
	/* copy up to next space */
	search_tags = 0;
	firstspace = q;
	*q++ = *p;
}
```

Disabling tags searching, variable `firstspace` linked to `q` and pointing to semicolon in metric.
`firstspace` is not NULL any more. Next iterations hits only this cases:
```
} else if (
		*p != '\0' && (
			firstspace != NULL ||
			(*p >= 'a' && *p <= 'z') ||
			(*p >= 'A' && *p <= 'Z') ||
			(*p >= '0' && *p <= '9') ||
			strchr(self->allowed_chars, *p)
		)
	)
{
	/* copy char */
	*q++ = *p;
} else {
	/* something barf, replace by underscore */
	*q++ = '_';
}
```
But `firstspace` is still linked to `q` and at the end of cycle it's look like `;host=ex1;cpu=2 4 1579606658` what cause aggregator later to mess up.